apps/frontend/src/app/5_pages/App/App.tsx

```diff
@@ -19,15 +26,52 @@ function App() {
   const { connectWallet, disconnectWallet, wallets, pending } =
     useWalletConnect();
 
+  const { setTransactions, setIsOpen, setTitle } = useTransactionContext();
+
+  const approve = useCallback(async () => {
+    if (!wallets[0].provider) return;
+    const { address, abi } = await getTokenContract('xusd', defaultChainId);
+    const provider = new ethers.providers.Web3Provider(wallets[0].provider);
+    const signer = provider.getSigner();
+    const xusd = new ethers.Contract(address, abi, signer);
+
+    setTransactions([
+      {
+        title: 'Approve XUSD tokens',
+        subtitle: 'Allow Sovryn protocol to use XUSD tokens for the trade',
+        contract: xusd,
+        fnName: APPROVAL_FUNCTION,
+        args: [ethers.constants.AddressZero, parseUnits('2')],
+      },
+      {
+        title: `Transfer 2 XUSD tokens`,
+        contract: xusd,
+        fnName: 'transfer',
+        args: [ethers.constants.AddressZero, parseUnits('2')],
+      },
+    ]);
+    setTitle('Transaction approval');
+    setIsOpen(true);
```

Comment:
Will be approving, but I think dialog should opened when you push transactions using `setTransactions`, so you wouldn't need to use `setIsOpen` each time.